### PR TITLE
perf(storagenode): remove backup from append request

### PR DESCRIPTION
**Pull request by ijsong**
### What this PR does

This PR removes the Backups fields from AppendRequest. Thus, it makes unmarshaling AppendRequest can
be improved.

### CODE REVIEW

**Comment by ijsong** [comment]
Current dependencies on/for this PR:
* main
  * **PR #409** <a href="https://app.graphite.dev/github/pr/kakao/varlog/409" target="_blank">Graphite</a> 
    * **PR #411** <a href="https://app.graphite.dev/github/pr/kakao/varlog/411" target="_blank">Graphite</a> 
      * **PR #412** <a href="https://app.graphite.dev/github/pr/kakao/varlog/412" target="_blank">Graphite</a>  👈
        * **PR #413** <a href="https://app.graphite.dev/github/pr/kakao/varlog/413" target="_blank">Graphite</a> 
    * **PR #410** <a href="https://app.graphite.dev/github/pr/kakao/varlog/410" target="_blank">Graphite</a> 

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/kakao/varlog/412?utm_source=stack-comment).

File: internal/storagenode/client/log_client.go [hunk 1]
```diff
@@ -41,12 +41,11 @@ func (c *LogClient) reset(rpcConn *rpc.Conn, _ types.ClusterID, target varlogpb.
 // Append stores data to the log stream specified with the topicID and the logStreamID.
 // The backup indicates the storage nodes that have backup replicas of that log stream.
 // It returns valid GLSN if the append completes successfully.
-func (c *LogClient) Append(ctx context.Context, tpid types.TopicID, lsid types.LogStreamID, data [][]byte, backups ...varlogpb.StorageNode) ([]snpb.AppendResult, error) {
+func (c *LogClient) Append(ctx context.Context, tpid types.TopicID, lsid types.LogStreamID, data [][]byte) ([]snpb.AppendResult, error) {
```
Comment (hungryjang) [inline internal/storagenode/client/log_client.go:44]: Is it correct that backups are not used because excutor has replica meta already?

Comment (ijsong) [reply internal/storagenode/client/log_client.go:44]: @hungryjang 
Yes, it is. A log stream replica gets to know other replicas by Unseal RPC.

**Review by hungryjang — APPROVED** [review]

**Comment by ijsong** [comment]
@ijsong started a stack merge that includes this pull request via [Graphite](https://app.graphite.dev/github/pr/kakao/varlog/412).

**Comment by ijsong** [comment]
@ijsong merged this pull request with [Graphite](https://app.graphite.dev/github/pr/kakao/varlog/412).

**Pull request merged**